ci/playwright_test.py

```diff
@@ -13,50 +13,71 @@
 # limitations under the License.
 # ==============================================================================
 
+import logging
 import re
 import time
 from playwright.sync_api import Page, expect
-import cv2
+from PIL import Image, ImageChops
 from pathlib import Path
 
 LOCAL_SERVER = "http://127.0.0.1:8080//"
 ROOT_DIR = Path(__file__).parent.parent
 TEST_FILES_DIR = ROOT_DIR / "test/test_models"
 TMP_SCREENSHOT_DIR = ROOT_DIR / "build"
+DEBUG_SCREENSHOT_DIR = TMP_SCREENSHOT_DIR / "debug"
 EXPECTED_SCREENSHOT_DIR = ROOT_DIR / "test/screenshots_golden/chrome-linux"
 
 
+# Compares two images, the similarity is determined by calculating the percentage
+# of mismatched pixels. `threshold` can be set between 0 to 1, 0 means the images
+# are identical.
 def matched_images(
-    actual_image_path, expected_image_path, threshold_ratio: int = 0.4
+    actual_image_path: Path, expected_image_path: Path, threshold: float = 0.005
```

Comment:
40/(1280*720)=0.000043, so maybe set threshold this to this number?

Reply:
Done!